examples/backpack-demo/index.html

```diff
@@ -0,0 +1,156 @@
+      <value name="A">
+        <shadow type="math_number">
+          <field name="NUM">1</field>
+        </shadow>
+      </value>
+      <value name="B">
+        <shadow type="math_number">
+          <field name="NUM">1</field>
+        </shadow>
+      </value>
+    </block>
+    <block type="math_single">
+      <value name="NUM">
+        <shadow type="math_number">
+          <field name="NUM">9</field>
+        </shadow>
+      </value>
+    </block>
+    <block type="math_number_property">
+      <value name="NUMBER_TO_CHECK">
+        <shadow type="math_number">
+          <field name="NUM">0</field>
+        </shadow>
+      </value>
+    </block>
+  </category>
+  <category name="Text" categorystyle="text_category">
+    <block type="text"></block>
+    <block type="text_multiline"></block>
+    <label text="Input/Output:" web-class="ioLabel"></label>
+    <block type="text_print">
+      <value name="TEXT">
+        <shadow type="text">
+          <field name="TEXT">abc</field>
+        </shadow>
+      </value>
+    </block>
+    <block type="text_prompt_ext">
+      <value name="TEXT">
+        <shadow type="text">
+          <field name="TEXT">abc</field>
+        </shadow>
+      </value>
+    </block>
+  </category>
+  <sep></sep>
+  <category name="Variables" categorystyle="variable_category" custom="VARIABLE"></category>
+  <category name="Functions" categorystyle="procedure_category" custom="PROCEDURE"></category>
+</xml>
+
+<script>
+  // Inject primary workspace.
+  var primaryWorkspace = Blockly.inject('primaryDiv',
+      {
+        media: 'https://unpkg.com/blockly/media/',
+        toolbox: document.getElementById('toolbox'),
+        trashcan: true,
+      });
+  // Inject secondary workspace.
+  var secondaryWorkspace = Blockly.inject('secondaryDiv',
+      {
+        media: 'https://unpkg.com/blockly/media/',
+        toolbox: document.getElementById('toolbox'),
+        trashcan: true
+      });
+
+  // Add backpacks
+  const backpack1 = new Backpack(primaryWorkspace);
+  backpack1.init();
+  const backpack2 = new Backpack(secondaryWorkspace);
+  backpack2.init();
+
+  // Listen to events on both workspace.
+  primaryWorkspace.addChangeListener(updateBackpack);
+  secondaryWorkspace.addChangeListener(updateBackpack);
+
+  function updateBackpack(event) {
+    if (event.type !== 'backpack_change') {
+      return;
+    }
+    Blockly.Events.disable();
+    let contents;
+    let targetBackpack;
+    if (primaryWorkspace.id === event.workspaceId) {
+      targetBackpack = secondaryWorkspace.backpack;
+      contents = primaryWorkspace.backpack.getContents();
+      console.log('second workspace backpack updated');
+    } else { // secondaryWorkspace.id === event.workspaceId
+      targetBackpack = primaryWorkspace.backpack;
+      contents = secondaryWorkspace.backpack.getContents();
+      console.log('first workspace backpack updated');
+    }
+    targetBackpack.setContents(contents);
```

Comment:
I'm thinking about if someone wanted to create a backpack that only lets the owner of the workspace delete items in the backpack, but lets the other users add items. I think this is possible now, but would it be easier if we added information on the event as to what changed? I'm not sure it is worth it, just curious : )

Reply:
I think adding more information to the change event could be useful.
I'll add it to the design doc and if it doesn't get sorted out before it is published, then I'll file a bug to keep track.
Since I'm still playing around with it, I'm not sure what kind of information is useful. So, I want to wait to add properties so there's minimal changing of properties on the event (only adding).

Reply:
Yup, sounds great : )